tests/rest/client/utils.py

```diff
@@ -142,14 +151,25 @@ def create_room_as(
         else:
             return None
 
-    def invite(self, room=None, src=None, targ=None, expect_code=200, tok=None):
-        self.change_membership(
+    def invite(
+        self,
+        room=None,
+        src=None,
+        targ=None,
+        expect_code=200,
+        tok=None,
+        appservice_user_id=None,
+        msc3316_ts=None,
```

Comment:
rather than having to extend all of these methods with new parameters specific to your test case, I'd recommend inlining them, and calling `send_request` directly. (Or possibly `HomeserverTestCase.send_request`, which sets the `reactor` and `site` parameters for you.)

So, instead of calling `self.helper.invite(room, appservice_user, normal_user, tok=appservice.token, msc3316_ts=ts, appservice_user_id=appservice_user)`, you would do:

```python
        url_params = {
            "user_id": appservice_user,
            "org.matrix.msc3316.ts": ts,
        }
        channel = self.make_request(
            "PUT",
            path=f"/_matrix/client/r0/rooms/{room}/state/m.room.member/{normal_user}?"+url_encode(url_params),
            content={"membership": "invite"},
            access_token=appservice.token,
        )
        self.assertEqual(channel.code, 200)
        event_id = channel.json_body["event_id"]
```

To my mind, this is much clearer than going via a set of utility functions whose effects are hard to understand.

The same applies to `create_room_as`, `send`, etc.

Reply:
I ended up rolling these all back *except* `create_room_as`, it should be easy enough to call that manually though, if you prefer?